File: 2019/day-04/part1.rb
def process(data)
  (data.first..data.last).count do |i|
    monotonic = true
    has_pair = false

    4.downto(0) do |j|
      pair = (i / (10 ** j)) % 100
      has_pair = pair % 11 == 0 unless has_pair
      monotonic = (pair % 10) >= (pair / 10)
      break unless monotonic
    end

    monotonic && has_pair
  end
end

def input
  File.read('./input.txt').split('-').map(&:to_i)
end

puts process(input)

File: 2015/day-05/part1.py
import itertools
import re

def input():
  return open('./input.txt', 'r').read().splitlines()

def process(data):
  count = 0

  for d in data:
    if len(re.findall(r'[aeiou]', d)) >= 3:
      if re.search(r'([a-z])\1', d):
        if len(re.findall(r'ab|cd|pq|xy', d)) == 0:
          count += 1

  return count

print(process(input()))
